### PR TITLE
Add nodejson driver

### PR DESCRIPTION
**Pull request by neko-para**
Add basic support for nodejson driver.
Add new dependence @types/node in driver.
Add LICENSE and tsconfig.

---

Still WIP.
Some behaviors need to be determined.

### CODE REVIEW

File: drivers/kurimudb-driver-nodejson/index.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const path = "./kurimudb.json"; // TODO: Maybe read it from model?
+let data = {};
+let loaded = false;
+
+function load(): void {
+  if (loaded) {
+    return;
+  }
+  loaded = true;
+  if (existsSync(path)) { // TODO: Find a way to use async methods.
```
Comment (jctaoo) [inline drivers/kurimudb-driver-nodejson/index.ts:12]: Here is a solution:
```typescript
export async function exists(path: fs.PathLike): Promise<boolean> {
  try {
    await fs.promises.stat(path);
    return true;
  } catch {
    return false;
  }
}

```

File: drivers/kurimudb-driver-nodejson/index.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+function save(): void {
+  writeFileSync(path, JSON.stringify(data));
+}
+
+export class NodeJsonDriver {
+    model;
+    root;
+  
+    constructor(model) {
```
Comment (jctaoo) [inline drivers/kurimudb-driver-nodejson/index.ts:29]: Maybe add a type-annotation will be better. Also, line 26 and 27

**Review by jctaoo — COMMENTED** [review]
Nice work!! Here are some suggestions.

File: drivers/kurimudb-driver-nodejson/index.ts (hunk 1 above)
Comment (neko-para) [reply drivers/kurimudb-driver-nodejson/index.ts:12]: Yes, I know we can use try-catch and stat to replace existSync. Actually I'm not sure if there would be a place for async initialize. Currently I can only init(aka load the json) in driver's constructor, which cannot be async.

**Comment by neko-para** [comment]
I'm not sure how model work in driver. Can user pass anything to driver(like dexie driver does)? I havn't found any example yet.
And is there a place for initialization? Or if it have to be initialized by user(e.g. add a class for json loading and saving, and pass it to driver)

**Comment by akirarika** [comment]
Thanks for your help!!

This commit let me know some of the flaws that drivers api design, I will improve them in the near future and write a detailed document~

file_328450

I’ll merge it first, I’m very very happy to be able to get your help quq!!

**Pull request merged**